test/server/cards/02_SHD/leaders/TheMandalorianSwornToTheCreed.spec.ts

```diff
@@ -0,0 +1,125 @@
+describe('The Mandalorian, Sworn To The Creed', function () {
+    integration(function (contextRef) {
+        describe('The Mandalorian\'s leader ability', function () {
+            beforeEach(function () {
+                contextRef.setupTest({
+                    phase: 'action',
+                    player1: {
+                        hand: ['green-squadron-awing', 'waylay'],
+                        leader: 'the-mandalorian#sworn-to-the-creed',
+                    },
+                    player2: {
+                        groundArena: ['wampa', 'battlefield-marine']
+                    },
+                });
+            });
+
+            it('should not exhaust units when not playing upgrades', function () {
```

Comment:
Similar comment as above - feels like this could be combined with the below test